ni/measurements/data/v1/data_store.proto

```diff
@@ -19,186 +19,129 @@ option objc_class_prefix = "NIMD";
-    string test_plan_id = 16;
-  }
-  // Optional. Represents the product associated with this session.
-  oneof product {
-    // The product associated with this session.
-    ni.measurements.metadata.v1.ProductMetadata product_metadata = 17;
-    // The id of the product associated with this session.
-    // This value is expected to be a parsable GUID or an alias. It will always
-    // be returned from the service as a GUID.
-    // See ni.measurements.metadata.MetadataStoreService for more information.
-    string product_id = 18;
-  }
-
-  // Optional. A link to a resource that describes the session.
+  string test_result_name = 15;
+  // Optional. Start date and time of the test execution (timestamp)
+  ni.protobuf.types.PrecisionTimestamp start_date_time = 16;
+  // Optional. End date and time of the test execution (timestamp)
+  ni.protobuf.types.PrecisionTimestamp end_date_time = 17;
+  // Optional. Overall test result outcome (e.g., Passed, Failed, Indeterminate)
+  Outcome outcome = 18;
+  // Optional. A link to a resource that describes the test result.
   // This value is expected to be a valid URI.
   string link = 19;
+  // Optional. Any extension to be associated with the test result.
+  map<string, ni.measurements.metadata.v1.ExtensionValue> extensions = 20;
+  // Optional.  If any of your messages contain 'extension', you must register
+  // a schema, and specify the schema id here.  In that case, all metadata will
+  // be validated against the specified schema.
+  string schema_id = 21;
 }
 
-// This represents a single published condition in a measurement.
-message Condition {
-  // The identifier describing the values of this condition. For example, "Voltage".
-  string name = 1;
-  // The type of this condition. For example, "Upper Limit" or "Environment".
-  string type = 2;
-  // The value for this condition for a single publish on the measurement.
-  oneof values {
-    ni.protobuf.types.Scalar scalar = 3;
-  }
-}
-
-// This represents all published values for a single condition in a measurement.
-// TODO: Rename to ConditionBatch after condition set is removed
-message ConditionArray {
-  // The identifier describing the values of this condition. For example, "Voltage".
-  string name = 1;
-  // The type of this condition. For example, "Upper Limit" or "Environment".
-  string type = 2;
-  // The values for this condition across for all publishes on the measurement.
-  oneof values {
-    ni.protobuf.types.Vector vector = 3;
-  }
-}
-
-// This represents all published conditions for a single measurement.
-message ConditionSet {
-  // The list of conditions (and all their values) that were published for a single measurement.
-  repeated ConditionArray condition_arrays = 1;
-}
-
-message PublishableData {
-  // Required for data. This name is used for associating/grouping conceptually alike data across multiple publish iterations.
+message Measurement {
```

Comment:
I was expecting `Measurement` and `MeasurementBatch` to get collapsed into the request messages similar to `Condition` and `ConditionBatch`.

Reply:
@nick-beer are you also happy with this?

Reply:
Yes, I'd expect the two to be consistent.

Reply:
In particular I am wondering if the Measurement and Condition messages will serve with the eventual addition of multiple entity publish methods

Reply:
If the batch messages are simply the repeated of the non-batch request message this won't be an issue.

Reply:
right, at this level I imagine it would be most performant to make the fields repeated as opposed to bundling it in a message.  Makes sense

Reply:
I think now that I'm looking at the modified version, that the oneofs will still need to be their own message types in order to become repeated fields, so the Measurement(Batch) and Condition(Batch) messages may still need to exist to serve that purpose, but with the "name" and "type" fields flattened into the request.

Reply:
Ah! yes, you're right. Alternatively, PublishMeasurementsRequest could just contain `repeated PublishMeasurementRequest` - right?

Reply:
My preference would still be to reserve specific meaning to the term "Batch" across all of our APIs like the AIPs I linked previously. However, I seem to be in the minority here so I'll let Nick make the final call on this. I think it might still be instructive to define terminology for what we would call a cell of data, a row of data, a column of data, and a range of rows/columns as you provided examples of earlier. If there were clear terms for those, maybe defining RPC names would be easier here. The closest I came was calling each cell a MeasurementValue or ConditionValue, each column MeasurementValues or ConditionValues, and anything involving rows a batch of the previous. However, that didn't seem to land with others.